### PR TITLE
Implement subtraction of intervals.

### PR DESCRIPTION
**Pull request by kevincox**
PR to pyinter: https://github.com/intiocean/pyinter/pull/11

@jamesemoody @angelini @datwright

### CODE REVIEW

File: pyinter/interval.py [hunk 1]
```diff
@@ -230,9 +230,28 @@ def union(self, other):
         else:
             return IntervalSet((self, other))
 
+    __or__ = __add__ = union
+
+    def __sub__(self, other):
+        """Returns a new Interval or an :class:`~pyinter.IntervalSet` representing the union of this
```
Comment (angelini) [inline pyinter/interval.py]: union?

**Comment by datwright** [comment]
what's [1,2) - (1,2)?

**Comment by datwright** [comment]
[1,1] ?

**Comment by kevincox** [comment]
Should be, want to to write a test for that case?

**Comment by kevincox** [comment]
Test added! All works as expected.

File: tests/test_interval.py [hunk 2]
```diff
@@ -63,6 +63,41 @@ def test_union_via_overloaded_or():
...
+    left  = i.open(3, 6)
+    right = i.open(4, 5)
+    expected = IntervalSet([i.openclosed(3, 4), i.closedopen(5, 6)])
+    assert left - right == expected
+
+def test_subtract_almost_complete_overlap():
+    left  = i.closed(1, 2)
+    right = i.open(1, 5)
+    expected = i.closed(1, 1)
+    assert left - right == expected
```
Comment (datwright) [inline tests/test_interval.py:98]: cool :+1:

Comment (datwright) [inline tests/test_interval.py:98]: what about [1,2] - (1,2)?

or anything that divides the interval should probably break, I think?

(1,4) - (2,3)

Comment (kevincox) [inline tests/test_interval.py:98]: Anything that divides the interval returns an `IntervalSet` with the two halves.

**Comment by kevincox** [comment]
Good to merge? This is blocking me.

**Comment by angelini** [comment]
LGTM

**Pull request merged**